kartothek/io_components/read.py

```diff
@@ -49,6 +49,13 @@ def dispatch_metapartitions_from_factory(
         )
         dispatch_by = dataset_factory.partition_keys
 
+    if dispatch_by and not set(dispatch_by).issubset(
+        set(dataset_factory.index_columns)
```

Comment:
Why cast to `set` here if you have already done so in the method `index_columns`?

Reply:
1. Because it doesn't matter and this way works no matter what index_columns returns
2. This code is older than the index_columns code ;)